src/lib/media/manip.ts

```diff
@@ -240,3 +251,64 @@ function normalizePath(str: string, allPlatforms = false): string {
   }
   return str
 }
+
+export async function saveBytesToDisk(
+  filename: string,
+  bytes: Uint8Array,
+  type: string,
+) {
+  const encoded = Buffer.from(bytes).toString('base64')
+  return await saveToDevice(filename, encoded, type)
+}
+
+export async function saveToDevice(
+  filename: string,
+  encoded: string,
+  type: string,
+) {
+  try {
+    if (isIOS) {
+      const tmpFileUrl = await withTempFile(filename, encoded)
+      await Sharing.shareAsync(tmpFileUrl, {UTI: type})
+      safeDeleteAsync(tmpFileUrl)
```

Comment:
@haileyok  Be careful here. although you do delete the tmpFile, the tmpDirUri will *not* be deleted!

The tmp file/dir will also not be deleted if Sharing.shareAsync fails.

You might want to do:

```ts
  const tmpDirUri = joinPath(documentDirectory as string, String(uuid.v4()))
  await makeDir
  try {
    await writeFileInDir
    await shareFile
    return true
  } finally {
    deleteDirAsync
  }
```